StrangeIoC/scripts/strange/framework/impl/Binder.cs

```diff
@@ -349,43 +349,51 @@ virtual public void ConsumeBindings(string jsonString)
 		{
 			List<object> list = Json.Deserialize(jsonString) as List<object>;
 			IBinding testBinding = GetRawBinding ();
-			int bindConstraints = (testBinding.keyConstraint == BindingConstraintType.ONE) ? 0 : 1;
-			bindConstraints |= (testBinding.valueConstraint == BindingConstraintType.ONE) ? 0 : 2;
 
 			for (int a=0, aa=list.Count; a < aa; a++)
 			{
-				Dictionary<string, object> item = list[a] as Dictionary<string, object>;
-				List<object> keyList;
-				List<object> valueList;
-				IBinding binding = null;
+				ConsumeItem(list[a] as Dictionary<string, object>, testBinding);
+			}
+		}
+
+		virtual protected IBinding ConsumeItem(Dictionary<string, object> item, IBinding testBinding)
```

Comment:
Very well covered but I think some of these methods are worthy of comment/summary.